tests/src/test/scala/jms4s/JmsClientSpec.scala

```diff
@@ -248,4 +251,225 @@ class JmsClientSpec extends AsyncFreeSpec with AsyncIOSpec with Jms4sBaseSpec {
+        for {
+          _ <- messages.parTraverse_(msg => {
+                producer.send(messageFactory(msg, outputQueueName1)) *> producer.send(
+                  messageFactory(msg, outputQueueName2)
+                )
+              })
+          _                   <- logger.info(s"Pushed ${messages.size} messages.")
+          _                   <- logger.info(s"Consumer to Producer started.\nCollecting messages from output queue...")
+          firstBatch          <- Ref.of[IO, Set[String]](Set())
+          firstBatchMessages  <- receiveUntil(outputConsumer, firstBatch, nMessages).timeout(timeout) >> firstBatch.get
+          secondBatch         <- Ref.of[IO, Set[String]](Set())
+          secondBatchMessages <- receiveUntil(outputConsumer2, secondBatch, nMessages).timeout(timeout) >> secondBatch.get
+        } yield assert(firstBatchMessages == bodies && secondBatchMessages == bodies)
+    }
+  }
+
+  s"send $nMessages messages in two Topics with unidentified pooled producer and consume them" in {
+    val jmsClient = new JmsClient[IO]
+
+    val res = for {
+      connection      <- connectionRes
+      session         <- connection.createSession(SessionType.AutoAcknowledge)
+      topic           <- Resource.liftF(session.createTopic(topicName))
+      outputConsumer  <- session.createConsumer(topic)
+      topic2          <- Resource.liftF(session.createTopic(topicName2))
+      outputConsumer2 <- session.createConsumer(topic2)
+      bodies          = (0 until nMessages).map(i => s"$i")
+      messages        <- Resource.liftF(bodies.toList.traverse(i => session.createTextMessage(i)))
+      producer <- jmsClient.createProducer(
+                   connection,
+                   poolSize
+                 )
+    } yield (producer, outputConsumer, outputConsumer2, bodies.toSet, messages)
+
+    res.use {
+      case (producer, outputConsumer, outputConsumer2, bodies, messages) =>
+        for {
+          _ <- messages.parTraverse_(msg => {
+                producer.send(messageFactory(msg, topicName)) *> producer.send(messageFactory(msg, topicName2))
+              })
+          _                   <- logger.info(s"Pushed ${messages.size} messages.")
+          _                   <- logger.info(s"Consumer to Producer started.\nCollecting messages from output queue...")
+          firstBatch          <- Ref.of[IO, Set[String]](Set())
+          firstBatchMessages  <- receiveUntil(outputConsumer, firstBatch, nMessages).timeout(timeout) >> firstBatch.get
+          secondBatch         <- Ref.of[IO, Set[String]](Set())
+          secondBatchMessages <- receiveUntil(outputConsumer2, secondBatch, nMessages).timeout(timeout) >> secondBatch.get
+        } yield assert(firstBatchMessages == bodies && secondBatchMessages == bodies)
+    }
+  }
+
+  s"sendN $nMessages messages with delay in a Queue with pooled producer and consume them" in {
+    val jmsClient = new JmsClient[IO]
+
+    val body  = "body"
+    val delay = 200.millis
```

Comment:
there's already a default `delay` in scope of 500 millions. 
Can we just use that?
Or we can even lower our default `delay` to 200ms..